pkgs/by-name/ld/ldc/generic.nix

```diff
@@ -14,52 +30,46 @@ let
 
 in
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "ldc";
   inherit version;
 
-  src = fetchurl {
-    url = "https://github.com/ldc-developers/ldc/releases/download/v${version}/ldc-${version}-src.tar.gz";
-    inherit sha256;
+  src = fetchFromGitHub {
+    owner = "ldc-developers";
+    repo = "ldc";
+    rev = "v${version}";
+    inherit hash;
+    fetchSubmodules = true;
   };
 
   # https://issues.dlang.org/show_bug.cgi?id=19553
   hardeningDisable = [ "fortify" ];
 
   postUnpack = ''
```

Comment:
I think that you should move this to postPatch or something else that happens after the builder moves into `sourceRoot`.
This way you don't have rely on the fact that `fetchFromGitHub` has `source` as its derivation name.

Reply:
Done, thanks